src/task/cleanup.php

```diff
@@ -26,28 +25,32 @@
             continue;
         }
 
-        $withinPath = sprintf('{{deploy_path}}/releases/%s', $release);
-        within($withinPath, function () use ($sudo, $release, $cleanupPaths, $withinPath): void {
-            $resolvedWithinPath = parse($withinPath);
-
-            foreach ($cleanupPaths as $cleanupPath) {
-                $absoluteRemovePath = sprintf('%1$s/%2$s', $resolvedWithinPath, $cleanupPath);
-                if (false === test(sprintf('[ -e %s ]', $cleanupPath))) {
-                    writeln(sprintf('Skipped "<comment>%1$s</comment>". The path does not exist.', $absoluteRemovePath));
-                    continue;
-                }
-
-                writeln(sprintf('Removing "%1$s"', $absoluteRemovePath));
-
-                try {
-                    run(sprintf('%s rm -rf %s', $sudo, $cleanupPath));
-                    writeln(sprintf('Removed "<info>%1$s</info>".', $absoluteRemovePath));
-                } catch (RunException $exception) {
-                    writeln(sprintf('Failed to remove "<comment>%1$s</comment>". %2$s', $absoluteRemovePath, $exception->getErrorOutput()));
-                } catch (Exception $exception) {
-                    writeln(sprintf('Failed to remove "<comment>%1$s</comment>".', $absoluteRemovePath));
-                }
+        $releasePath = sprintf('{{deploy_path}}/releases/%1$s', $release);
```

Comment:
just wondering, instead of going through all releases, wouldn't it make more sense to just cleanup the previous release (if it exists `previous_release`)? as all others would have been cleaned already with previous iterations (at least once the job has been in use for a while).